packages/andromeda_protocol/src/gumball.rs

```diff
@@ -0,0 +1,109 @@
+use common::{
+    ado_base::{recipient::Recipient, AndromedaMsg, AndromedaQuery},
+    mission::AndrAddress,
+};
+use cosmwasm_std::{Addr, Binary, Coin, Uint128};
+use cw721_base::MintMsg;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::cw721::TokenExtension;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    pub andromeda_cw721_contract: AndrAddress,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    AndrReceive(AndromedaMsg),
+    Mint(Box<MintMsg<TokenExtension>>),
+    Buy {},
+    // Sets price, max amount per wallet, and recipient
+    SaleDetails {
```

Comment:
nit: Since this is an execute call and is making mutating changes, we should prefix this with `Set` or something of that sort. On its own this sounds like a query.